### PR TITLE
4254 fix race gin process wait

### PR DESCRIPTION
**Pull request by akaspin**
The shutdown [issue](https://trello.com/c/9E4vHyvE/4254-act-tracker-used-after-close) wasn't fixed by https://github.com/remerge/go-service/pull/63. The issue is that due it's nature Go HTTP Server may accept request after close on connection that still open. Here are the fix:

1. This PR blocking process requests after server enters in closing state.
1. Wait for processing invokes after both servers reported that they are closed.

### CODE REVIEW

**Comment by kolo** [comment]
I checked `graceful` package and noticed that the project is archived and the author [recommends using built-in `http.Server`](https://github.com/tylerb/graceful#using-go-18). What do you think about assessing this and getting rid of `graceful`? It might be a separate ticket though.

**Comment by akaspin** [comment]
@kolo Agree with author because after code inspection I can't see any purpose to use this package :-D Of course in separate ticket.

**Comment by kolo** [comment]
Here you go https://trello.com/c/BhDaaAzw/546-go-service-use-built-in-httpserver-instead-of-gracefulserver

File: server.go [hunk 1]
```diff
@@ -149,6 +139,19 @@ func (s *Server) Shutdown(os.Signal) {
...
+	atomic.StoreUint32(&s.closing, 1)
+	allRequestsServedChan := make(chan struct{})
+	go func() {
+		s.requestsWg.Wait()
+		close(allRequestsServedChan)
+	}()
+	select {
+	case <-allRequestsServedChan:
+		s.log.Info("all requests processed")
+	case <-time.After(s.ShutdownTimeout):
```
Comment (kolo) [inline server.go]: `ShutdownTimeout` is a total timeout for all services, if there are other services before `Server` we might never reach it.

Comment (akaspin) [reply server.go]: Actually it's _Server's_ `ShutdownTimeout`. This timeout also set in `graceful.Server` and has own command-line parameter: https://github.com/remerge/go-service/blob/bee8df20db36a805688497c6ad3a796ae0efbfbd/server.go#L74-L78.

**Pull request merged**